### PR TITLE
fix: Allow PATCH to update value from 'non-null' to 'null'

### LINKED ISSUE #5875: Fail on updating user profile image
**Describe the bug**
The current flow of updating the user profile image inside Open Event Android is:
1. Upload the new encoded image string to the server by
`POST  /v1/upload/image` (This works fine)
2. After updating the image string, the server will return the image URL to Android application (This works fine)
3. Android application continue to update user profile with
`PATCH  /v1/users/{user_id}` (This returns 500 INTERNAL SERVER ERROR)

Please take a look

**Stacktrace**
```
V/InputMethodManager: Reporting focus gain, without startInput
V/AudioManager: playSoundEffect   effectType: 0
D/OkHttp: --> POST https://api.eventyay.com/v1/upload/image
D/OkHttp: Content-Type: application/json; charset=UTF-8
    Content-Length: 7104
    Authorization: JWT eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1NTc3ODMyNzYsImlhdCI6MTU1NzY5Njg3NiwibmJmIjoxNTU3Njk2ODc2LCJpZGVudGl0eSI6NzYyMn0.TfG7lbkevyVBdFzEyBpvUPzy8WT2eabH_CgdgSSou3U
D/OkHttp: {"data":"data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEB\nAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/2wBDAQEBAQEBAQEBAQEBAQEBAQEBAQEB\nAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/wAARCAB4AHgDASIA\nAhEBAxEB/8QAHwABAAIDAAMAAwAAAAAAAAAAAAkKBwgLAQMGAgQF/8QANxAAAgICAgEDAwIEAwcF\nAAAABAUDBgIHAAEICRMUERIVCiIWFxgkGSEyIzlYh6e31yUxNDVW/8QAFAEBAAAAAAAAAAAAAAAA\nAAAAAP/EABQRAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwEAAhEDEQA/AOf/AMcccBxxxwHHMlao05tT\nelvEoWn6Badi20zGGb8NVlJLOQACZkvUZOnhUWHwa9XBGLZaO0sr4pbX1HzYJWrIODP3eTaaM/T/\nAG+Lgvhcb42rTtL4EihlQVWuLM9qW8efsqfBgqsMwjisU5PPGLHDMGwr1ovQs2RP2zwwZD54SBAD\nxy6BQPQl8Fqc0lYWOLcG1xZB+4cUV92EOtUjy99Zf3cEur67rd53P13l13jgQ6IF/Zj1kPl+/wC/\nMX+Dl6cP/Dn/ANXd7/8Ak/gUXeOXQL/6EvgtcWkTCuR7g1QLGN1DkjoWwR2ameXrrH+8nl2hXdjv\nPf77677ywHdji9/dl1iPh+z7Iu95/p/t9U5fM40RtSnbqwGEKJnq1hW56rt5BHzIYgVdekNc2amt\n5OxJZSTWFhtdHHi7GywHHIznjjxCAPjmSNq6e2lo23l0Pb9BtOuraHjNN2mtSgpXMcBExPU4uUxE\n2HQb+vFsFbEdZZEZLBC2+HPKrYlw4e5zG/AccccBxxxwHHHHAclx9On0pr75owfzQvjdpq3x8BZ4\ngi2MddFNa9nlr2WI9gV67gY49rw1aqMc9Wy2CzFapl1n6jSKUNuPVW8SsefSm9OeDzRvbe/bQ7ZA\n+PmrWq8axCA9MQDdn2yWLFlDrxW/G6gwUqgl+QjPYLNafjZgE7ZCpRRqz7YNb6tdTSJE1aTKK5XF\nCxBXkCwBIhQpABVSZImVCxAq1ChWBFACtWLQYIAwAA4IRQxYYhx4o4Y8MMQxjo/QGnPG2kQ650hQ\nUuvqjEaUzmXq8jjDWbQ3Lrsho+fuS2VhsbPuHCAKJi+asjR1Ya9SPPEsXACD5h4565poR4ZSCJY4\nIII85p55s8YoYYYse85JZZM+8cI448Mcs8888uscMeu8su+uuu++B7OOVU/NP119gubEzo3hnEJS\nakkcFDd7osKRbYLPd4A8hY4S6xT7UoMR1KuGTxs+8M7Ire2ZyqITnZDUNlCxTyQ9/wBc3mj+R/Kf\n1Z+R3yfm/P8Aa/nRsPtd7/v/ACPb/Ed2H8T8L3P2/jfhfjvY/tfi/G/2XA6GHHKrHhR66t/VWRXQ\n/MyMK5VV44EFj3UgTK67Y6VEbIXHMVbKlWFQSC0VsWaVX1lPXFaCwpFYzY7MW9MCAVo9qXDPCTDC\nSPPGSOTHHPDPDLrLDPDLrrLHPDLHvvHLHLHvrvHLrvvrvrvrvrvvrvgYc3n4+aZ8laRNrvd9BTX+\nqSGDMRxGORoTFSzEkxzhZ1+wpy11hrjLvDqQMg5E1XkmKyT05shCpieETT59RP0ptg+F0Gez6O3Y\nbT8fDW8gM1llXYQWzWUzBj3BXFexxgMfx5S9ngSCnX7BVDrEjSzY5qmiGmnuqgtsV2vn8x2kTWVM\n3rljULLBXrAsPSPkLsARqmdpmokoDRQ3VnxTgsljIGecM8AyCYQwSaUciKSGTPDIOZ1xyWr1WfTv\ng8LtiqLrrCJyb497SOYY1nA6Jgxz1nbIOszi9bNLJN2TixBIX+821+wdGR2VsjXvVbLF6fS29ufR\nK8BxxxwHMl6a1Ncd7bVoGnaAH0Zbti2hVV0/vQM5wF+bEjHAx877TLm7IStVtf0XYLO0GWG/h68s\nZtph84ApeY05YG/T+6HCt279q+QDoWKeDTtTXVSoYFqysusLjs3tpCwfJ3PREQkR1fpteeV9muyH\nMllA2IORlmF1HB2WFnTQekqR44adoOktdQGQ1DXyTpSulYkdFM2ZZJZTZ6+az4YQwZtrG/YNH7TE\nIYJdGwZExrAF6/AYKDL3HHAciZ9andLvT/gxalte+cOy3TcK5paRuvP6BlUJH4D+12r3sPjzdsAb\nFVaU6pLADGQXuQK0zkdkd4jZDESzcib9afSzjcPgvbWKD55DPS1urm6e1K8HAyRsnr4T2q2v5OeU\n8PYAFdqN0fXU47DAnPAaryD+x9hOU8IUi+OOOA5d79F/djXcvgvTAHvbAltpWz2DScrVhOPN+UVV\noJJZaj0FGPBB8VfXqTca3TRoJ/eIy6rWZUs8nycfpSE5d39FvSjfTngvT2L7547XdVrsO68lLAOI\nXJQosQSOr1P4kkRBHZwFiptMr11DMlxGl9q0dCZD9dC4yzBLDxxxwMK+RWiqZ5MaU2Ho6/YEdVvY\nCPtdIaHLPEclbgliuazZAe4CBfeNrVkXKXwoRMma1lMuxXNxjVJRoRHPR2vrO16Y2Ze9T3gTAO26\n8tTqovY4OivhTnJDpgsmCuU0UIgtK0jijZpGGYkGLJSWEwij6hJj+vSY5UW9fzRa6keRGtN5qIwh\nY960s9VZR4pmk7Ay6al/BJiLAb0XLMvEGOpFjoSBeGp+NF3nU2BpYfRhkxpwQIccccBy6V6FmvTa\nX4ILrIUeOZBtzbGxdhLR4Ys45FQS3NNqiQArPLPPGciRjrE9pjLHjFhiKyGg7w7khzkkpa8vRejl\n/u4PHP8A5u/999n8CTbjjjgOeuaGIiKUciKOeCePOGaGbDGWKaKXHvCSKWPPrLCSOTDLLDPDPHvH\nPHvvHLrvrvvrns44FVHzT9CnYaWxs7z4aSCXeovHJRP8mH7tZX7RRhzOxpYhK1b7U4DR26thkZs8\nY8bE0R2dOrhTAZT3tjKydRw+deDPmjkw6WdeJnkf8ns3oDqXvS2xMV/v9z/H9ztvlXulWIXufu7Y\n5G9L+oP7nsrob/a86GPHAqseFXoU35tYll78zZQ6bVUjcUrDStedLLDZLpEFIZnMLa7fV2piGrVw\nuaJX33DWmj+xO1RbYLIyiMRgWU9qSOOOKPCKLDCKKLDGOOOPHHCOOPDHrHDDDDHrrHDDDHrrHHHH\nrrHHHrrrrrrrrn58cBxxxwHIV/XkobW2eES+zKxBpoNYboo1vsJUvWGJIldbqrXr36C594d55dEW\ne61bCeDHLDCTDHGbP7sho+upqORk+sb/ALuDyM/5Rf8AffWHAou8cccBy6B6Et/aXHwWirjCIeMX\nVG4NhUJHlD9OpZ1LIeu7QlnL+mOPfc/TzY7ofDLvvP6jDj49Z/TH7MKX/J/f0/8AvUCnb52loVzO\nMPFumnrrFVZiSje5p7jqztwbNX1YEUEofuuKZY7TYGJxMoveA9FFGwynkniixC23xxzWrzD8gYfF\nrxn2/vjJb23NodZwzr6zMXIwM24WRstqFJhcQRslBHde7tz9J3ZJQ2EDAdB0yIXdTnRDjyh+l5P+\nZXj14gVeOx7tvQyY5iMwlq1IUQZvL9cygV55uAderQffvRjlzAfiMbM/nR0lY5OVg2C0Ju2Iskle\nvfv6gjab4hkn8btR1jX6TPGyLhrls2ae6XQkUr2oKzY1dcUFJqlT3qyLok41M6I2qkIOmDHynICA\nJxbwL7N2bfdy320bQ2faGdzvtzZyN7HY28keRZxeUcQ8EUUA8UAK1YtBgFVpEisUJMhT
    BAJUoACk\nAMOD4Xgbr7A9R7zr2Y0FcWPym2+tLED6Bih1/ZpdTK84cZM5esykeq4qakNM+6TLrtiavIYZR9YQ\n5E9wxxx44d/qi8mfz38Vf1E71/if/wDSfzcv/wCe/wBP2/8A3H8QfkP9P7f/AJH+n/L/ANv8uYK4\n4G7OvvUg87NZMzG1c8pdusijg/gzRbBsme2lmEPu4TfeGl2rDc0y8z78MevyIAAzD2u84Pk+xJJH\nlKhoH9QTstKSCn8ldQVu8pMcK6vluWq5iKfcRBw4CoLI/bVd6c4qtxeOcu15oqxKx1WkXlRMo4/q\nGeFAnrrccDoe+MPmBoHy+qEts0jdoHkqyFX/ABXUGY8ia8Ug5oBCbGus1dL796PGOTMlbE/TzOKe\n4Zq3A9csjrBYXNHs3zmyav2jsDS99rWztW2trSr3UWMbNDYU8uGBAs+PWUc4xI88c4LRSxFkmXOk\nbYU5K9UlGKHIByw0oSboLeKO/FXlB47an3upC/GYbArGJjVV1iR1CotSY82tXROHIXhGQWuU29K8\nWrWEkeHbIAUZhjhjgTj10GwvIZfXbv7Sn+DGVbXYjZi7W3Dr+iPfe6x7mjTrBrHs+OYT645ZYzdP\ntdIoZM8e8O+hp54+8/pJ9mczXKjvr/b2X3Tf+sdDp5hSYtIU497aZoM2MZQ1z2r+FZ9IGI5EMIE3\nSqk12mv1xwGZWPeF0ODmnjIFnGiCAnjjjgOZJ07te36M2nQdwUIvEO266tKm0pvemZQgHSrCsJSU\njrFQwVMC67YQflIrKsHYh/l0DFkrlnwgMl5jbjgdHfQG76T5IaboG79dzGS1LYKT8qBCxH7FZKzR\nDClD9A0h+ucPTSt2Fc1QMpApi1s5y0idWeeukFNn+s2Prmkbdolq1nsiuAW2jXVOUistfY+/gOwX\nF49fX2yRJhj1x4s2MRqturLCbpmYwbZQcEzDFLhpmelN6iw/hdfXFD2h0zO8fNpNFxFjKBzZMC9Y\nWuCLpdBsRVXh8iI2qs0D4qzYS1WB3Z2KZSgapJGx1RDp9nuqpHaayplFjrjdY/rz9YA7Qvkh4rVM\n7TNRYjlbdQ0AlnBZLGQM8BgB4c8wpgs0RA8skMmGeQUUPUL9PPYng7sGWXCJtbNDWtvPBrTZso8c\nneOcsRB0NHvEwMEAKy9rgRis48sRwF1yXLjLDXRBsRH6KtR086UextcUTbtIset9mVVPdaNbQPx1\ngrT0XEtcwHwnhLGk+n1xmFPXHjCNFDUKUZmmbhAt1JgbMIQqGtn5degg4gOe3bw6toZyubNo1w0p\nsRnmIyA+vZhw6Gh7BJ6lDaw95yCpUa7YeaicIQfAyxbHclTTk9BWo45sptrw48qdF92CXavj/tSp\nKKvPAO7t01QataALmThDlBlBsVFA0ojGCXsiOHElbYjBvlfeJ3L0VFLDhrXwHHNl9TeGvlZvPKvS\n6r8fdq2tRapJ4kVujqDVRr8vsaOeSeSXYz+BVQwIMOxpYfksbEIPkX1GFhLkZNDBJOJ4jege0lMR\n3XzFt4gq6LtU1y0prlnISeb3/wCnsCEN+2FDhEKuiwzwPQP1mu+2cpg83Z9a2arnigIzCL308/Tx\n2H5xX/CeWNrUtC1NtBBsrZkQ8ceUksUYx01Fospo5ALO9sACRZZpMxz1lLWMA7DYhSuzK7X7ReK1\nxrmkajotV1nreug1OjUpOIhrVfXe/mOvXCY99Y9SElzEnsTipcpTWjdoWa3csyS2rc41kYUXN511\nrmi6kpNd1xrSqpqVR6oD2vr9aQB4BLV8EhExhMnWGP1kJOYnkltGzMyQhk4bGnNmhZjI0oqb6F27\nTVpM3sdjbrK/Xq+sPdvnzs8RUmSJlQkp7Ru3aHywArVi0GCcw88yeEQMSGUgiWOGPPPEMVeQ+9KX\n406W2DvHYEhPVY1+k/JTiAwzTnuWppgqet11fjDDN1GfZLIxUoRCyuolgE7HA9uWCqGMNH56e3No\n2zdmz77tu8lYF23Ylqc2x3lBmbkCKU4NlKxVqI2JrEwREmHzhUIF0xxXStKEAuimyhFj5JP6rPqI\nweaOw1FJ1hI4C8fNWnsM61IdKxXS7Mtk+OYJWyGlcn+NitAHX++p18vchZ2ZUjYvWjPJGfc2tRr8\nSnAccccBxxxwHJcfTo9Vm9+F0Her76na7S8fDmWJwlcGYxQ2vV5h7LEmwNdeTMe+gDVTWOc9my16\nzLUJmFnyjeKX9SOaXAq0RHccDo86R37p3yQo8GxtI35LsGoTGkrJWKr5ghixoH9mRCp8ibiLrBXG\nuMEoxuCx8rWnzLDV7SEeRcxBKIy/zm5an3JtXRNwDv2nr/aNdW4PGCHtxVmpK3NgBAzXOe0j0SPL\ntfY62WyULCWlYfiMq847CHiarDIMOouTbaM/UBb1qC+BPvnVFP3LgOKAJBa6yzk1bbp5cCZ8mbaw\njjKLPT3RMwskEYS+vVuhBwyDZdzSy/J7zgC2vxyGmgeuz4KXFpOvsU+4NUCQj+9G8v8AryBirKk/\nd/aQQ6tsWynWJH7ev3EqBxf34/3P+Wf2Zg/xjfTh/wCIz/pFvf8A8YcCTbjkMt/9dvwYpzTFdW8t\nw7XGzF6n/O0TX4ytPFN3j130HNHs+xa6fdTY5d/bJJAiIG6+3LuOeTr7Pvi63n+oA37c18yfROra\nbpPAkScee0P2OW1bgMV0bDMIyr2LBLWqYqy7DikEMX2CoXaHP5MkoxQ8sUMnQWeN6+Q2l/Gmky7C\n3hsBLQKtiWOvFIY9GnNHDEmWOKJbXa4mFZWOyMMMZOyyg0SphOArgNcH4DKV5xo1Pr1EvVZ2H5ow\nSawpKg/Vvj4E4kOkrUjDGa2bNlXMffrjTZBQOfa8YBbiME5X6+UzsUaqzZ5NGj25npKi0r8bO09v\nbR3fbS73t2/WnYttLwkg7dWtwW2IDBkPOZ4J1EREmQqNAIczPnW15NAAjVfLmiWLxIc/b5jngOOO\nOA4444DjjjgOOOOA4444DjjjgOOOOA4444DjjjgOOOOB/9k=\n"}
    --> END POST (7104-byte body)
D/OkHttp: <-- 200 OK https://api.eventyay.com/v1/upload/image (458ms)
    Server: nginx/1.14.0 (Ubuntu)
    Date: Sun, 12 May 2019 22:03:24 GMT
    Content-Type: application/json
    Transfer-Encoding: chunked
D/OkHttp: Connection: keep-alive
    Vary: Accept-Encoding
    Access-Control-Allow-Origin: *
D/OkHttp: {"url":"https://api.eventyay.com/static/media/temp/images/6d7c271f-fa4a-47b2-9e03-cc0b637e0b0e/TlBHdTQ1ak/94e0bff4-3356-45b3-ab79-23cb08229df0.jpeg"}
D/OkHttp: <-- END HTTP (150-byte body)
D/OkHttp: --> PATCH https://api.eventyay.com/v1/users/7622
    Content-Type: application/vnd.api+json
    Content-Length: 581
    Authorization: JWT eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1NTc3ODMyNzYsImlhdCI6MTU1NzY5Njg3NiwibmJmIjoxNTU3Njk2ODc2LCJpZGVudGl0eSI6NzYyMn0.TfG7lbkevyVBdFzEyBpvUPzy8WT2eabH_CgdgSSou3U
D/OkHttp: {"data":{"type":"user","id":"7622","attributes":{"first-name":"Đức","last-name":"Lê","email":null,"contact":null,"details":null,"thumbnail-image-url":null,"icon-image-url":null,"small-image-url":null,"avatar-url":"https://api.eventyay.com/static/media/temp/images/6d7c271f-fa4a-47b2-9e03-cc0b637e0b0e/TlBHdTQ1ak/94e0bff4-3356-45b3-ab79-23cb08229df0.jpeg","facebook-url":null,"twitter-url":null,"instagram-url":null,"google-plus-url":null,"original-image-url":null,"created-at":null,"last-accessed-at":null,"deleted-at":null,"admin":false,"super-admin":false,"verified":false}}}
    --> END PATCH (581-byte body)
D/EditProfileViewModel$updateProfile: Image uploaded https://api.eventyay.com/static/media/temp/images/6d7c271f-fa4a-47b2-9e03-cc0b637e0b0e/TlBHdTQ1ak/94e0bff4-3356-45b3-ab79-23cb08229df0.jpeg
D/OkHttp: <-- 500 INTERNAL SERVER ERROR https://api.eventyay.com/v1/users/7622 (1035ms)
D/OkHttp: Server: nginx/1.14.0 (Ubuntu)
    Date: Sun, 12 May 2019 22:03:25 GMT
    Content-Type: application/vnd.api+json
D/OkHttp: Content-Length: 140
    Connection: keep-alive
    Access-Control-Allow-Origin: *
    X-Sentry-ID: 00b1fa5f792d4ad392532088f1e118d4
D/OkHttp: {"errors": [{"detail": "Unknown error", "source": {"pointer": ""}, "status": 500, "title": "Unknown error"}], "jsonapi": {"version": "1.0"}}
D/OkHttp: <-- END HTTP (140-byte body)
E/EditProfileViewModel$updateUser: Error updating user!
    retrofit2.adapter.rxjava2.HttpException: HTTP 500 INTERNAL SERVER ERROR
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:54)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:44)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:12267)
        at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual(ObservableSingleSingle.java:35)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:784)

```
**Additional details (please complete the following information):**
- Android: API 24, Honor 7

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5875 

#### Short description of what this resolves:
Allowing PATCH request to change a column value from `non-null` to `null`

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @shreyanshdwivedi @CosmicCoder96 @iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
OK, this will allow fields to be null. But there's a catch, we want only certain fields to behave this way. For example, for most fields, we want null to be resolved to default and only 1-2 fields to resolve to null

This is because in Android and may happen in other clients as well, we convert classes to JSON and send the data, where most data is null except the fields we want to PATCH

Second concern is, we got what will happen if the field is null. What will happen if the field is not present in JSON?

**Comment by prateekj117** [comment]
@iamareebjamal If the field is not present in JSON, it will not patch the field and let it be what it earlier was. Though, if we pass null for a field in JSON, it will update the field to null. 

In the case stated by you, I think we can hard code the fields in which we want null to be resolved to default, or change the implementation in Android if possible.

**Comment by iamareebjamal** [comment]
> I think we can hard code the fields in which we want null to be resolved to default

Instead, specify fields in which null should be resolved to null, as the previous behaviour was to resolve to default or skip. So, all nulls will be skipped unless otherwise told

File: utils/common.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def patch_defaults(schema, in_data):
     for name, field in schema.fields.items():
         dasherized_name = dasherize(name)
         attribute = in_data.get(dasherized_name)
-        if attribute is None and dasherized_name not in in_data:
+        if attribute is None and dasherized_name not in in_data and field.skip_none is False:
```
Comment (iamareebjamal) [inline utils/common.py]: Will fail for fields not having `skip_none`

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 Weirdly enough, I tried doing it in dev branch, it's now updating from avatar-url from non-null to null. Can you guys please  verify. 
@iamareebjamal Any idea why?

**Comment by prateekj117** [comment]
@iamareebjamal It worked in dev branch as well now.

This was the request body:
```
{
  "data": {
    "attributes": {
      "avatar_url": null
    },
    "type": "user",
    "id": "2"
  }
}
```

This was the response:
```
{
    "data": {
        "type": "user",
        "attributes": {
            "was-registered-with-order": false,
            "billing-phone": null,
            "facebook-url": null,
            "avatar-url": null,
            "last-name": "example",
            "google-plus-url": null,
            "is-verified": true,
            "company": null,
            "billing-country": null,
            "is-marketer": false,
            "is-super-admin": false,
            "deleted-at": null,
            "facebook-id": null,
            "icon-image-url": null,
            "is-user-organizer": false,
            "billing-address": null,
            "is-sales-admin": false,
            "billing-additional-info": null,
            "billing-zip-code": null,
            "is-user-coorganizer": false,
            "is-user-owner": false,
            "last-accessed-at": "2019-07-03T05:58:28.124515+00:00",
            "twitter-url": null,
            "is-user-moderator": false,
            "created-at": "2019-06-12T09:11:22.901375+00:00",
            "instagram-url": null,
            "thumbnail-image-url": null,
            "is-admin": false,
            "billing-tax-info": null,
            "email": "test1@example.com",
            "billing-city": null,
            "is-user-registrar": false,
            "details": "verified user, test1",
            "is-user-track-organizer": false,
            "contact": null,
            "original-image-url": null,
            "is-user-attendee": false,
            "billing-contact-name": null,
            "small-image-url": null,
            "first-name": "test1"
        },
        "relationships": {
            "email-notifications": {
                "links": {
                    "self": "/v1/users/2/relationships/email-notifications",
                    "related": "/v1/users/2/email-notifications"
                }
            },
            "marketer-events": {
                "links": {
                    "self": "/v1/users/2/relationships/marketer-events",
                    "related": "/v1/events"
                }
            },
            "speakers": {
                "links": {
                    "self": "/v1/users/2/relationships/speakers",
                    "related": "/v1/users/2/speakers"
                }
            },
            "orders": {
                "links": {
                    "self": "/v1/users/2/relationships/orders",
                    "related": "/v1/users/2/orders"
                }
            },
            "access-codes": {
                "links": {
                    "self": "/v1/users/2/relationships/access-codes",
                    "related": "/v1/users/2/access-codes"
                }
            },
            "favourite-events": {
                "links": {
                    "self": "/v1/users/2/relationships/favourite-events",
                    "related": "/v1/users/2/user-favourite-events"
                }
            },
            "attendees": {
                "links": {
                    "self": "/v1/users/2/relationships/attendees",
                    "related": "/v1/users/2/attendees"
                }
            },
            "sales-admin-events": {
                "links": {
                    "self": "/v1/users/2/relationships/sales-admin-events",
                    "related": "/v1/events"
                }
            },
            "sessions": {
                "links": {
                    "self": "/v1/users/2/relationships/sessions",
                    "related": "/v1/users/2/sessions"
                }
            },
            "owner-events": {
                "links": {
                    "self": "/v1/users/2/relationships/owner-events",
                    "related": "/v1/events"
                }
            },
            "registrar-events": {
                "links": {
                    "self": "/v1/users/2/relationships/registrar-events",
                    "related": "/v1/events"
                }
            },
            "coorganizer-events": {
                "links": {
                    "self": "/v1/users/2/relationships/coorganizer-events",
                    "related": "/v1/events"
                }
            },
            "moderator-events": {
                "links": {
                    "self": "/v1/users/2/relationships/moderator-events",
                    "related": "/v1/events"
                }
            },
            "organizer-events": {
                "links": {
                    "self": "/v1/users/2/relationships/organizer-events",
                    "related": "/v1/events"
                }
            },
            "track-organizer-events": {
                "links": {
                    "self": "/v1/users/2/relationships/track-organizer-events",
                    "related": "/v1/events"
                }
            },
            "alternate-emails": {
                "links": {
                    "self": "/v1/users/2/relationships/alternate-emails",
                    "related": "/v1/users/2/alternate-emails"
                }
            },
            "notifications": {
                "links": {
                    "self": "/v1/users/2/relationships/notifications",
                    "related": "/v1/users/2/notifications"
                }
            },
            "events": {
                "links": {
                    "self": "/v1/users/2/relationships/events",
                    "related": "/v1/users/2/events"
                }
            },
            "discount-codes": {
                "links": {
                    "self": "/v1/users/2/relationships/discount-codes",
                    "related": "/v1/users/2/discount-codes"
                }
            }
        },
        "id": "2",
        "links": {
            "self": "/v1/users/2"
        }
    },
    "links": {
        "self": "/v1/users/2"
    },
    "jsonapi": {
        "version": "1.0"
    }
}
```

**Comment by iamareebjamal** [comment]
And you had set `avatar-url` before? Meaning it was non-null first and then you used this payload and it set the `avatar-url` to null?

**Comment by prateekj117** [comment]
@iamareebjamal Yes.

**Comment by iamareebjamal** [comment]
Show this change. i.e., before PATCH user, after PATCH user

**Comment by prateekj117** [comment]
Before PATCH:
Screenshot from 2019-07-03 23-16-59

After PATCH:
Screenshot from 2019-07-03 23-17-25

**Comment by iamareebjamal** [comment]
I see no change at all

**Comment by iamareebjamal** [comment]
@anhanh11001 Can you try this?

**Comment by anhanh11001** [comment]
Yes, I'll do it later today

**Comment by iamareebjamal** [comment]
@anhanh11001 Update?

**Comment by anhanh11001** [comment]
oh sorry for forgetting about this. It works well on Android

**Comment by prateekj117** [comment]
@iamareebjamal Closing this.

**Comment by anhanh11001** [comment]
@prateekj117 why is this closed instead of getting merged

**Comment by prateekj117** [comment]
@anhanh11001 Because some other PR must have fixed this. I was able to change the value from `non-null` to `null` in the dev branch itself.

**Comment by iamareebjamal** [comment]
@anhanh11001 Because you said it was working fine on Android?

**Comment by anhanh11001** [comment]
Yes, it only worked in the dev branch. I was just confused

**Pull request closed without merging**